v7.1 changelog + docs (#5746)
mattermost/docs: source/install/self-managed-changelog.md

See the [changelog in progress](https://bit.ly/2nK3cVf) for the upcoming release. See the [Legacy Self-Hosted Mattermost Changelog](legacy-self-hosted-changelog) for details on all Mattermost self-hosted releases prior to v6.0. 

Latest Mattermost Releases:
- [Release v7.1 - Feature Release](#release-v7-1-feature-release)
- [Release v7.0 - Major Release](#release-v7-0-major-release)
- [Release v6.7 - Feature Release](#release-v6-7-feature-release)
- [Release v6.6 - Feature Release](#release-v6-6-feature-release)
- [Release v6.3 - Extended Support Release](#release-v6-3-extended-support-release)

## Release v7.1 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release Day: 2022-07-16**

### Important Upgrade Notes

>  - We create a new config option ``MaxImageDecoderConcurrency`` which indicates how many images can be decoded concurrently at once. The default is -1 which means number of CPUs present. This affects the total memory consumption of the server. The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, a good rule of thumb to follow is that ``MaxImageResolution* MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.

Reviewer: ```suggestion
 - A new configuration option ``MaxImageDecoderConcurrency`` indicates how many images can be decoded concurrently at once. The default is -1, and the value indicates the number of CPUs present. This affects the total memory consumption of the server. The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, a good rule of thumb to follow is that ``MaxImageResolution * MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.
```

Reviewer: What does the default value of -1 mean, if the value equals the number of CPUs present? Does -1 mean that this setting is disabled?

> **IMPORTANT:** If you upgrade from a release earlier than v7.0, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

Reviewer: ```suggestion
**IMPORTANT:** If you upgrade from a release earlier than v7.1, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

See the [changelog in progress](https://bit.ly/2nK3cVf) for the upcoming release. See the [Legacy Self-Hosted Mattermost Changelog](legacy-self-hosted-changelog) for details on all Mattermost self-hosted releases prior to v6.0. 

Latest Mattermost Releases:
- [Release v7.1 - Extended Support Release](#release-v7-1--extended-support-release)
- [Release v7.0 - Major Release](#release-v7-0-major-release)
- [Release v6.7 - Feature Release](#release-v6-7-feature-release)
- [Release v6.6 - Feature Release](#release-v6-6-feature-release)
- [Release v6.3 - Extended Support Release](#release-v6-3-extended-support-release)

## Release v7.1 - [Extended Support Release](https://docs.mattermost.com/upgrade/release-definitions.html#extended-support-release-esr)

**Release Day: 2022-07-16**

### Important Upgrade Notes

>  - A new configuration option ``MaxImageDecoderConcurrency`` indicates how many images can be decoded concurrently at once. The default is -1, and the value indicates the number of CPUs present. This affects the total memory consumption of the server. The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, a good rule of thumb to follow is that ``MaxImageResolution * MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.

Reviewer: ```suggestion
 - A new configuration option ``MaxImageDecoderConcurrency`` indicates how many images can be decoded concurrently at once. The default is -1, and the value indicates the number of CPUs present. This affects the total memory consumption of the server. The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, we recommend that ``MaxImageResolution * MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.
```

 - Adding an @mention at the start of a post draft and pressing the left or right arrow key can clear the post draft and the undo history [MM-33823](https://mattermost.atlassian.net/browse/MM-33823).
 - Google login fails on the Classic mobile apps.
 - Status may sometimes get stuck as **Away** or **Offline** in High Availability mode with IP Hash turned off.
 - Searching stop words in quotation marks with Elasticsearch enabled returns more than just the searched terms.
 - The team sidebar on the desktop app does not update when channels have been read on mobile.
 - Slack import through the CLI fails if email notifications are enabled.
 - Push notifications don't always clear on iOS when running Mattermost in High Availability mode.
 - Boards are not refreshing on creation. See the [GitHub discussion](https://github.com/mattermost/focalboard/discussions/1971) for more information.
 - Boards export and reimport results in duplicates boards because all IDs are replaced by new ones on the server. See the [GitHub issue](https://github.com/mattermost/focalboard/issues/1924) for more information.

> 

Reviewer 2: ```suggestion
- The Top Boards widget in Insights is slow to load.

```

>  - Boards export and reimport duplicates boards because all IDs are replaced by new ones on the server. See the [GitHub issue](https://github.com/mattermost/focalboard/issues/1924) for more information.

Reviewer: ```suggestion
 - Boards export and reimport results in duplicates boards because all IDs are replaced by new ones on the server. See the [GitHub issue](https://github.com/mattermost/focalboard/issues/1924) for more information.
```

- [Release v6.6 - Feature Release](#release-v6-6-feature-release)
- [Release v6.3 - Extended Support Release](#release-v6-3-extended-support-release)

## Release v7.1 - [Extended Support Release](https://docs.mattermost.com/upgrade/release-definitions.html#extended-support-release-esr)

**Release Day: 2022-07-16**

### Important Upgrade Notes
 - A new configuration option ``MaxImageDecoderConcurrency`` indicates how many images can be decoded concurrently at once. The default is -1, and the value indicates the number of CPUs present. This affects the total memory consumption of the server. The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, we recommend that ``MaxImageResolution * MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.
 - Mattermost v7.1 introduces schema changes in the form of a new column and its index. The following notes our test results for the schema changes:
    - MySQL 12M Posts, 2.5M Reactions - ~1min 34s (instance: PC with 8 cores, 16GB RAM)
    - PostgreSQL 12M Posts, 2.5M Reactions - ~1min 18s (instance: db.r5.2xlarge)

>  - Customers wanting to run the SQL queries before-hand can run these queries. This is fully backwards-compatible, but will obtain a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database.

Reviewer: ```suggestion
 - Customers wanting to run the SQL queries before-hand can run the following queries before upgrading. This is fully backwards-compatible and obtains a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database until the migrations are complete.
```
```suggestion
 - Customers wanting to run the SQL queries before-hand can run these queries. This is fully backwards-compatible, but will obtain a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database.
```

Latest Mattermost Releases:
- [Release v7.1 - Extended Support Release](#release-v7-1--extended-support-release)
- [Release v7.0 - Major Release](#release-v7-0-major-release)
- [Release v6.7 - Feature Release](#release-v6-7-feature-release)
- [Release v6.6 - Feature Release](#release-v6-6-feature-release)
- [Release v6.3 - Extended Support Release](#release-v6-3-extended-support-release)

## Release v7.1 - [Extended Support Release](https://docs.mattermost.com/upgrade/release-definitions.html#extended-support-release-esr)

**Release Day: 2022-07-16**

Mattermost v7.1.0 contains low severity level security fixes. [Upgrading](https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html) to this release is recommended. Details will be posted on our [security updates page](https://mattermost.com/security-updates/) 30 days after release as per the [Mattermost Responsible Disclosure Policy](https://mattermost.com/security-vulnerability-report/).

### Important Upgrade Notes

>  - A new configuration option ``MaxImageDecoderConcurrency`` indicates how many images can be decoded concurrently at once. The default is -1, and the value indicates the number of CPUs present. This affects the total memory consumption of the server. The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, we recommend that ``MaxImageResolution * MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.

Reviewer 3: > The maximum memory of a single image is dictated by ``MaxImageResolution * 24 bytes``. Therefore, we recommend that ``MaxImageResolution * MaxImageDecoderConcurrency * 24`` should be less than the allocated memory for image decoding.

This doesn't sound right. I assume we meant bits instead of bytes as 24-bit (true color) is the most standard bit depth used for images nowadays. With that said, nothing prevents users from uploading denser formats. The current supported limit would be (N)RGBA64 so up to 64 bits (or 8 bytes) per pixel.

Author: Adding @agnivade to confirm.

Reviewer 3: @jupenur Could you point me to where this allocation happens in the JPEG decoder? I am trying to understand if this in addition to the space allocated for the raw image or if it's included in it.

Reviewer 3: Okay so I am assuming they are separate as that data there seems to be purely allocated for processing. Still unsure where the 24 is coming from exactly though but I guess it should serve as an accurate enough estimate.

 - Mattermost v7.1 introduces schema changes in the form of a new column and its index. The following notes our test results for the schema changes:
    - MySQL 12M Posts, 2.5M Reactions - ~1min 34s (instance: PC with 8 cores, 16GB RAM)
    - PostgreSQL 12M Posts, 2.5M Reactions - ~1min 18s (instance: db.r5.2xlarge)

>  - Customers wanting to run the SQL queries before-hand can run the following queries before upgrading. This is fully backwards-compatible and obtains a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database until the migrations are complete.

Reviewer: ```suggestion
 - You can run the following SQL queries before the upgrade to obtain a lock on ``Reactions`` table, so that users' reactions posted during this time won't be reflected in the database until the migrations are complete. This is fully backwards-compatible.
```
